src/extensions/sb-ncbr/partial-charges/property.ts

```diff
@@ -0,0 +1,193 @@
+import { Model } from '../../../mol-model/structure';
+import { CustomProperty } from '../../../mol-model-props/common/custom-property';
+import { PropertyWrapper } from '../../../mol-model-props/common/wrapper';
+import { ParamDefinition as PD } from '../../../mol-util/param-definition';
+import { MmcifFormat } from '../../../mol-model-formats/structure/mmcif';
+import { CustomPropertyDescriptor } from '../../../mol-model/custom-property';
+import { CustomModelProperty } from '../../../mol-model-props/common/custom-model-property';
+
+type TypeId = number;
+type IdToCharge = Map<number, number>;
+type ChargesData = {
+    typeIdToMethod: Map<TypeId, string>;
+    typeIdToAtomIdToCharge: Map<TypeId, IdToCharge>;
+    typeIdToResidueToCharge: Map<TypeId, IdToCharge>;
+    maxAbsoluteAtomCharges: IdToCharge;
+    maxAbsoluteResidueCharges: IdToCharge;
+    maxAbsoluteAtomChargeAll: number;
+};
+type PartialCharges = PropertyWrapper<ChargesData | undefined>;
+
+let updatedParams = false;
+let PartialChargesPropertyParams = {
+    typeId: PD.Select<number>(0, [[0, '0']]),
+};
+type PartialChargesPropertyParams = typeof PartialChargesPropertyParams;
+let defaultPartialChargesPropertyParams = PD.clone(PartialChargesPropertyParams);
+
+function getParams(model: Model) {
+    // This is a hack for setting the option values for the dropdown menu
+    if (!updatedParams) {
+        const typeIdToMethod = getTypeIdToMethod(model);
+        const options = Array.from(typeIdToMethod.entries()).map(
+            ([typeId, method]) => [typeId, method] as [number, string]
+        );
+        PartialChargesPropertyParams = {
+            typeId: PD.Select<number>(1, options),
+        };
+        defaultPartialChargesPropertyParams = PD.clone(PartialChargesPropertyParams);
+        updatedParams = true;
+    }
+
+    return PartialChargesPropertyParams;
```

Comment:
This will only work for 1 model. Remove the `updatedParams` and just return a copy of the `PartialChargesPropertyParams` with the `typeId` as select.

If you wanted to optimize it, you can use `model._staticPropertyData` to cache the parsed structure. See for example https://github.com/molstar/molstar/blob/master/src/extensions/pdbe/struct-ref-domain.ts#L18 how we use this.